commit 1458a84f6756b5889a424aa1d8067c95a20f3bbf
Author: Mario Haustein <mario.haustein@hrz.tu-chemnitz.de>
Date:   Fri Apr 3 15:09:01 2026 +0200

    Link Alpaca focuser and filter wheel against httplib

diff --git a/drivers/filter_wheel/CMakeLists.txt b/drivers/filter_wheel/CMakeLists.txt
index e8ee05fd2..1d073d977 100644
--- a/drivers/filter_wheel/CMakeLists.txt
+++ b/drivers/filter_wheel/CMakeLists.txt
@@ -91,5 +91,5 @@ SET(alpaca_filterwheel_SRC
     indi_alpaca_filterwheel.cpp)
 
 add_executable(indi_alpaca_filterwheel ${alpaca_filterwheel_SRC})
-target_link_libraries(indi_alpaca_filterwheel indidriver)
-install(TARGETS indi_alpaca_filterwheel RUNTIME DESTINATION bin)
\ No newline at end of file
+target_link_libraries(indi_alpaca_filterwheel indidriver ${HTTPLIB_LIBRARY})
+install(TARGETS indi_alpaca_filterwheel RUNTIME DESTINATION bin)
diff --git a/drivers/focuser/CMakeLists.txt b/drivers/focuser/CMakeLists.txt
index aa6f5e733..95c96ea0d 100644
--- a/drivers/focuser/CMakeLists.txt
+++ b/drivers/focuser/CMakeLists.txt
@@ -405,5 +405,5 @@ SET(alpaca_focuser_SRC
     indi_alpaca_focuser.cpp)
 
 add_executable(indi_alpaca_focuser ${alpaca_focuser_SRC})
-target_link_libraries(indi_alpaca_focuser indidriver)
-install(TARGETS indi_alpaca_focuser RUNTIME DESTINATION bin)
\ No newline at end of file
+target_link_libraries(indi_alpaca_focuser indidriver ${HTTPLIB_LIBRARY})
+install(TARGETS indi_alpaca_focuser RUNTIME DESTINATION bin)
