Upstream: https://github.com/JonathanSalwan/Triton/pull/1246
--- a/src/libtriton/includes/triton/pythonBindings.hpp
+++ b/src/libtriton/includes/triton/pythonBindings.hpp
@@ -9,11 +9,6 @@
 #define TRITONPYTHONBINDINGS_H
 
 #include <Python.h>
-#if __has_include(<cpython/longintrepr.h>)
-  #include <cpython/longintrepr.h>
-#else
-  #include <longintrepr.h>
-#endif
 
 #if defined(_WIN32) && !defined(__WINE__)
   #include <cmath>
