diff --git a/src/libtriton/CMakeLists.txt b/src/libtriton/CMakeLists.txt
index 325ff5d1..06199799 100644
--- a/src/libtriton/CMakeLists.txt
+++ b/src/libtriton/CMakeLists.txt
@@ -350,7 +350,7 @@ target_link_libraries(
 add_dependencies(check triton)
 
 # Add the install targets
-install (TARGETS ${PROJECT_LIBTRITON} DESTINATION lib)
+install (TARGETS ${PROJECT_LIBTRITON} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install (DIRECTORY ${TRITON_ROOT}/src/libtriton/includes/triton DESTINATION include)
 install (DIRECTORY ${CMAKE_BINARY_DIR}/src/libtriton/includes/triton DESTINATION include)
 
