Upstream: https://github.com/JonathanSalwan/Triton/pull/1247
--- a/src/libtriton/CMakeLists.txt
+++ b/src/libtriton/CMakeLists.txt
@@ -278,6 +278,8 @@ if(WIN32 AND BUILD_SHARED_LIBS)
     target_compile_definitions(triton PUBLIC TRITON_BUILDING_DLL)
 endif()
 
+include(GNUInstallDirs)
+
 # Install tritonTargets.cmake
 install(
     EXPORT tritonTargets
@@ -289,10 +291,10 @@ install(
 install(
     TARGETS triton
     EXPORT tritonTargets
-    PUBLIC_HEADER DESTINATION include/triton
-    INCLUDES DESTINATION include
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/triton"
+    INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
 # Configure tritonConfig.cmake and tritonConfigVersion.cmake
@@ -300,7 +302,7 @@ include(CMakePackageConfigHelpers)
 configure_package_config_file(
     Config.cmake.in
     "${PROJECT_BINARY_DIR}/tritonConfig.cmake"
-    INSTALL_DESTINATION lib/cmake/triton
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/triton"
     NO_SET_AND_CHECK_MACRO
     NO_CHECK_REQUIRED_COMPONENTS_MACRO
 )
@@ -314,7 +316,7 @@ install(
     FILES
         "${PROJECT_BINARY_DIR}/tritonConfig.cmake"
         "${PROJECT_BINARY_DIR}/tritonConfigVersion.cmake"
-    DESTINATION lib/cmake/triton
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/triton"
 )
 
 # Install Python bindings
