diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98f80df..15ce470 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,8 +63,6 @@ endif(BUILD_VERILOG)
 
 # avr toolchain
 find_program(AVR_CC avr-gcc)
-# gzip
-find_program(GZIP gzip)
 
 # valgrind tool to detect memory problems
 find_program(VALGRIND valgrind)
@@ -103,7 +101,7 @@ find_program(DPKG dpkg)
 find_program(STRIP strip)
 find_program(FAKEROOT fakeroot)
 
-if(DPKG AND GZIP AND STRIP AND FAKEROOT)
+if(DPKG AND STRIP AND FAKEROOT)
   message(STATUS "can build debian package")
   set(BUILD_DEB ON)
 else()
@@ -112,9 +110,6 @@ else()
   if(NOT DPKG)
     message(STATUS "  - dpkg not found")
   endif()
-  if(NOT GZIP)
-    message(STATUS "  - gzip not found")
-  endif()
   if(NOT STRIP)
     message(STATUS "  - strip not found")
   endif()
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 965002f..2212d66 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -11,60 +11,44 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ChangeLog ${CMAKE_CURRENT_BINARY_DIR}
 
 add_custom_target(progdoc COMMENT "create program documentation files")
 
-if(HELP2MAN AND GZIP)
+if(HELP2MAN)
   message(STATUS "can create man page")
 
-  set(DOC_MAN ${CMAKE_CURRENT_BINARY_DIR}/simulavr.1.gz)
+  set(DOC_MAN ${CMAKE_CURRENT_BINARY_DIR}/simulavr.1)
   
   add_custom_command(OUTPUT ${DOC_MAN}
-                     COMMAND ${HELP2MAN} -o ${CMAKE_CURRENT_BINARY_DIR}/simulavr.1 "$<TARGET_FILE:simulavr>"
-                     COMMAND ${GZIP} -n9 ${CMAKE_CURRENT_BINARY_DIR}/simulavr.1)
+                     COMMAND ${HELP2MAN} -o ${CMAKE_CURRENT_BINARY_DIR}/simulavr.1 "$<TARGET_FILE:simulavr>")
   add_custom_target(doc-man DEPENDS ${DOC_MAN})
   add_dependencies(progdoc doc-man)
   
   install(FILES ${DOC_MAN} DESTINATION "share/man/man1")
 else()
   message(STATUS "can't create man page because of:")
-  if(NOT GZIP)
-    message(STATUS "  - gzip not found")
-  endif()
-  if(NOT HELP2MAN)
-    message(STATUS "  - help2man not found")
-  endif()
+  message(STATUS "  - help2man not found")
 endif()
 
-if(MAKEINFO AND GZIP)
+if(MAKEINFO)
   message(STATUS "can create info page")
 
-  set(DOC_INFO ${CMAKE_CURRENT_BINARY_DIR}/simulavr.info.gz)
+  set(DOC_INFO ${CMAKE_CURRENT_BINARY_DIR}/simulavr.info)
 
   add_custom_command(OUTPUT ${DOC_INFO}
-                     COMMAND ${MAKEINFO} ${CMAKE_CURRENT_BINARY_DIR}/simulavr.texi
-                     COMMAND ${GZIP} -n9 ${CMAKE_CURRENT_BINARY_DIR}/simulavr.info)
+                     COMMAND ${MAKEINFO} ${CMAKE_CURRENT_BINARY_DIR}/simulavr.texi)
   add_custom_target(doc-info DEPENDS ${DOC_INFO})
   add_dependencies(progdoc doc-info)
 
   install(FILES ${DOC_INFO} DESTINATION "share/info")
 else()
   message(STATUS "can't create info page because of:")
-  if(NOT GZIP)
-    message(STATUS "  - gzip not found")
-  endif()
-  if(NOT MAKEINFO)
-    message(STATUS "  - makeinfo not found")
-  endif()
+  message(STATUS "  - makeinfo not found")
 endif()
 
-if(GZIP)
-  set(DOC_CHANGELOG ${CMAKE_CURRENT_BINARY_DIR}/changelog.gz)
+set(DOC_CHANGELOG ${CMAKE_CURRENT_BINARY_DIR}/changelog)
 
-  add_custom_command(OUTPUT ${DOC_CHANGELOG}
-                     COMMAND ${GZIP} -n9 ${CMAKE_CURRENT_BINARY_DIR}/changelog)
-  add_custom_target(doc-changelog DEPENDS ${DOC_CHANGELOG})
-  add_dependencies(progdoc doc-changelog)
+add_custom_target(doc-changelog DEPENDS ${DOC_CHANGELOG})
+add_dependencies(progdoc doc-changelog)
 
-  install(FILES ${DOC_CHANGELOG} DESTINATION "share/doc/simulavr-1.1.0")
-endif()
+install(FILES ${DOC_CHANGELOG} DESTINATION "share/doc/simulavr-1.1.0")
 
 add_custom_target(doc COMMENT "create simulavr documentation" DEPENDS progdoc)
 
@@ -89,4 +73,4 @@ if(DOXYGEN AND DOT)
 
 endif()
 
-# EOF
\ No newline at end of file
+# EOF
