diff --git a/meson.build b/meson.build
index 35bbbb3..5c26a9c 100644
--- a/meson.build
+++ b/meson.build
@@ -7,9 +7,7 @@ version : '5.7.8',
   default_options : ['c_std=c11', 'cpp_std=c++11']
 )
 
-pugixml_sources = [
-  'third-party/pugixml/src/pugixml.cpp'
-]
+pugixml_dep = dependency('pugixml')
 
 r2ghidra_sources = [
   'src/ArchMap.cpp',
@@ -28,7 +26,6 @@ r2ghidra_sources = [
 
 incdirs = [
   'src',
-  'third-party/pugixml/src/',
   'ghidra-native/src/decompiler/',
 ]
 
@@ -180,7 +180,6 @@ ghidra_decompiler_sources = [
 
 r2ghidra_core_sources = [
   r2ghidra_sources,
-  pugixml_sources,
   ghidra_decompiler_sources,
   'src/anal_ghidra_plugin.c',
   'src/anal_ghidra.cpp',
@@ -190,7 +196,6 @@ r2ghidra_core_sources = [
 
 sleighc_sources = [
   r2ghidra_sources,
-  pugixml_sources,
   'ghidra-native/src/decompiler/slgh_compile.cc',
   'ghidra-native/src/decompiler/slghparse.cc',
   'ghidra-native/src/decompiler/slghscan.cc',
@@ -199,7 +194,7 @@ sleighc_sources = [
 
 r2ghidra_core_plugin = library('core_r2ghidra',
   r2ghidra_core_sources,
-  dependencies: [r_core],
+  dependencies: [r_core, pugixml_dep],
   override_options : ['c_std=c11', 'cpp_std=c++11'],
   include_directories: r2ghidra_incdirs,
   install: true,
@@ -209,6 +208,6 @@ r2ghidra_core_plugin = library('core_r2ghidra',
 sleighc_exe = executable('sleighc', sleighc_sources,
   include_directories: r2ghidra_incdirs,
   override_options : ['c_std=c11', 'cpp_std=c++11'],
-  dependencies: [r_core],
+  dependencies: [r_core, pugixml_dep],
   install: true
 )
