diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 4f95c72..24e74d8 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -146,33 +146,33 @@ GR_LIBRARY_FOO(gnuradio-satnogs)
 message(STATUS "Using install prefix: ${CMAKE_INSTALL_PREFIX}")
 message(STATUS "Building for version: ${VERSION} / ${LIBVER}")
 
-########################################################################
-# Build and register unit test
-########################################################################
-include(GrTest)
-
-# If your unit tests require special include paths, add them here
-include_directories(${ITPP_INCLUDE_DIRS})
-# List all files that contain Boost.UTF unit tests here
-list(APPEND test_satnogs_sources
-    qa_conv_coding.cc
-    qa_crc.cc
-    qa_golay24.cc
-    qa_reed_muller.cc
-    qa_utils.cc
-    qa_whitening.cc
-)
-
-# Anything we need to link to for the unit tests go here
-list(APPEND GR_TEST_TARGET_DEPS gnuradio-satnogs)
-
-if(NOT test_satnogs_sources)
-    MESSAGE(STATUS "No C++ unit tests... skipping")
-    return()
-endif(NOT test_satnogs_sources)
-
-foreach(qa_file ${test_satnogs_sources})
-    GR_ADD_CPP_TEST("satnogs_${qa_file}"
-        ${CMAKE_CURRENT_SOURCE_DIR}/${qa_file}
-    )
-endforeach(qa_file)
+#########################################################################
+## Build and register unit test
+#########################################################################
+#include(GrTest)
+#
+## If your unit tests require special include paths, add them here
+#include_directories(${ITPP_INCLUDE_DIRS})
+## List all files that contain Boost.UTF unit tests here
+#list(APPEND test_satnogs_sources
+#    qa_conv_coding.cc
+#    qa_crc.cc
+#    qa_golay24.cc
+#    qa_reed_muller.cc
+#    qa_utils.cc
+#    qa_whitening.cc
+#)
+#
+## Anything we need to link to for the unit tests go here
+#list(APPEND GR_TEST_TARGET_DEPS gnuradio-satnogs)
+#
+#if(NOT test_satnogs_sources)
+#    MESSAGE(STATUS "No C++ unit tests... skipping")
+#    return()
+#endif(NOT test_satnogs_sources)
+#
+#foreach(qa_file ${test_satnogs_sources})
+#    GR_ADD_CPP_TEST("satnogs_${qa_file}"
+#        ${CMAKE_CURRENT_SOURCE_DIR}/${qa_file}
+#    )
+#endforeach(qa_file)
