diff --git a/setup.py b/setup.py
index b7316f15..5fea3ba8 100644
--- a/setup.py
+++ b/setup.py
@@ -127,8 +127,7 @@ def default_build():
 
     setup(
         name="chirp",
-        packages=["chirp", "chirp.drivers", "chirp.ui", "tests", "tests.unit",
-                  "chirp.wxui"],
+        packages=["chirp", "chirp.drivers", "chirp.ui", "chirp.wxui"],
         version=CHIRP_VERSION,
         scripts=["chirpw", "rpttool", "chirpwx.py"],
         data_files=[('share/applications', desktop_files),
