From 99719048096f35fd4ba99ec693b5fb03e9e00679 Mon Sep 17 00:00:00 2001
From: mcgillca <colin.a.mcgill@gmail.com>
Date: Tue, 1 Oct 2024 20:21:03 +0100
Subject: [PATCH] Updated to include <cstdint>

Now defined u_int8 and 16.
--- a/mcp3421.cpp
+++ b/mcp3421.cpp
@@ -55,6 +55,9 @@ int mcp3421::openMCP3421()
     }
     else
         return nErr;
+
+    return nErr;
+
 }
 
 int mcp3421::closeMCP3421()
--- a/mcp3421.h
+++ b/mcp3421.h
@@ -2,6 +2,7 @@
 #include <iostream>
 #include <fcntl.h>
 #include <unistd.h>
+#include <cstdint>
 #include <sys/ioctl.h>
 
 #define ADC_ADDR0 0x68
