diff --git a/PCA9685.cpp b/PCA9685.cpp
index 17d4b405..b508509a 100644
--- a/PCA9685.cpp
+++ b/PCA9685.cpp
@@ -186,12 +186,10 @@ int PCA9685::setOff(uint8_t nPort)
 
 bool PCA9685::isPortOn(uint8_t nPort)
 {
-    int nErr = 0;
     int on_value = 0;
     int off_value = 0;
 
-    nErr = getPWM(nPort, on_value, off_value);
-    return (on_value==4096);
+    getPWM(nPort, on_value, off_value);
 }
 
 
