diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index a35ab8b..b2b2c04 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -41,18 +41,7 @@ endif(ENABLE_DOXYGEN)
 ########################################################################
 if(ENABLE_MANPAGES)
 
-FIND_PROGRAM(BZIP
-             NAMES bzip2
-             PATHS /bin
-                   /usr/bin
-                   /usr/local/bin)
-
-if(NOT BZIP)
-        MESSAGE(FATAL_ERROR "Could not find bzip2 for man page compression.")
-endif(NOT BZIP)
-
 set(MAN_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR})
-set(MAN_BUILD_DIR ${PROJECT_BINARY_DIR}/docs)
 
 list(APPEND manpages
      gr_satellites.1
@@ -64,11 +53,8 @@ list(APPEND manpages
 set(MAN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/share/man)
 
 foreach(manpage ${manpages})
-    add_custom_target(generate_manpage_${manpage} ALL
-        COMMAND ${BZIP} -c ${MAN_SRC_DIR}/${manpage} > ${MAN_BUILD_DIR}/${manpage}.bz2
-    )
     install(FILES
-       ${MAN_BUILD_DIR}/${manpage}.bz2
+       ${MAN_SRC_DIR}/${manpage}
        DESTINATION ${MAN_INSTALL_DIR}/man1
     )
 endforeach(manpage)
