From b0e7f51ae996e19a772940dd0b0c9d4e6d9333d3 Mon Sep 17 00:00:00 2001
From: Jasem Mutlaq <mutlaqja@ikarustech.com>
Date: Sat, 2 Dec 2023 11:00:38 +0300
Subject: [PATCH] Add missing header

--- a/indi_libcamera.cpp
+++ b/indi_libcamera.cpp
@@ -24,6 +24,7 @@
 
 #include <stream/streammanager.h>
 #include <indielapsedtimer.h>
+#include <sharedblob.h>
 
 #include "image/image.hpp"
 #include "core/still_options.hpp"
